src/modules/avformat/producer_avformat.c

```diff
@@ -138,9 +143,19 @@ struct producer_avformat_s
 #endif
 	int autorotate;
 	int is_audio_synchronizing;
+#ifdef USE_VAAPI
+	AVBufferRef* hw_device_ctx;
+	AVFrame* sw_video_frame;
+#endif
 };
 typedef struct producer_avformat_s *producer_avformat;
 
+#ifdef USE_VAAPI
+#define HW_PIX_FMT AV_PIX_FMT_VAAPI
+#define HW_DEVICE_TYPE AV_HWDEVICE_TYPE_VAAPI
+#define HW_DEVICE "/dev/dri/renderD128"
```

Comment:
Can these be properties that match with ffmpeg usage: `hwaccel`, `hwaccel_device`, and `hwaccel_output_format` perhaps with some defaults when unspecified per the `hwaccel`? If these would need to be supplied during producer initialization, they can be supplied as query string arguments on the resource property. `hwaccel` could be the way to explicitly choose to use this.